devlock.c

```diff
@@ -91,6 +91,8 @@ static gboolean device_autolock_timeout_cb(gpointer data)
 	(void)data;
```

Comment:
Please split that into separate commits:<br>1.the one with debug prints<br>2.functional changes commit.<br>Also, please elaborate (with details) in the commit message what is the problem and how it is solved. Why it works on n900 and not on d4?

Reply:
1. Id rather not split it.

2. the current code is supposed to work like this: a event reaches https://github.com/maemo-leste/mce/blob/3cbee89d364eea0211fa7a37493e39a6c2b22192/event-input.c#L168 here the event gets processed the callback then uses a timer to suspend the iomonitor for 1 second: https://github.com/maemo-leste/mce/blob/3cbee89d364eea0211fa7a37493e39a6c2b22192/event-input.c#L200 during this time n events pile up in the evdev device. When the timer times out it will re-enable the iomonitor and mce_resume_io_monitor will seek to the end of the file, effectively skiping the n events that piled up.

3. This, however, is broken because you can not seek evdev devices so the seek fails. Upon resume of the iomonitor touchscreen_cb will instead simply get the next event that piled up during the 1 second suspend and will then suspend the iomonitor again. this will repeat until every event is processed. Because the droid4's touchscreen creates events at a rate of 400Hz while swiping this will take a _very_ _very_ long time. During this time mce will think the device is being interacted with so the device will enter device_idle state only after a very long time.

4. this pr solves this problem by supplying the io_monitor with the function mcs_io_monitor_seek_to_end. This function attempts to seek to the end of a file. If it can not it will instead read() to the end of the file and discard the data.

Reply:
regarding why this works on n900. I dont know and cant confirm that it dose. sicelo on irc claims it works. assumeing it dose:

1. maybe the kernel allows seeking in some instances (why?)
2. maybe this never worked but the update rate of the n900 touchscreen is low enough that this is not so appearant.

Reply:
What is the error seek fails with?

Reply:
I had written a long answer etc., however, github somehow managed to lose it :( . Anyway, in short it was - lets wait until me or @MerlijnWajer is able to test on n900 (and maybe pinephone) to see if seek works there or not. If it works, I think we shall understand why (and why it doesn't on d4).